File: 2-1.php
<?php

$filename = 'pandas/edu_bigdata_imp1.csv';

$csvData = array_map('str_getcsv', file($filename));
$header = array_shift($csvData);

$pseudoIDIndex = array_search('PseudoID', $header);
$videoItemSNIndex = array_search('dp001_video_item_sn', $header);
$indicatorIndex = array_search('dp001_indicator', $header);

$uniqueValues = [];

foreach ($csvData as $row) {
    if ($row[$pseudoIDIndex] == 281) {
        $uniqueValues[] = [$row[$videoItemSNIndex], $row[$indicatorIndex]];
    }
}

$uniqueValues = array_map("unserialize", array_unique(array_map("serialize", $uniqueValues)));

foreach ($uniqueValues as $row) {
    echo implode(', ', $row) . PHP_EOL;
}

?>

File: 2-1.py
import pandas as pd

df = pd.read_csv('pandas\edu_bigdata_imp1.csv', encoding = 'big5', low_memory = False)
df_filtered = df[df['PseudoID'] == 281]
unique_value = df_filtered[['dp001_video_item_sn', 'dp001_indicator']].drop_duplicates()
print(unique_value)
